File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ds = {}
        for i in range(len(nums)):
            if target-nums[i] in ds.keys():
                return [i,ds[target-nums[i]]]
            else:
                ds[nums[i]] = i
        return [-1,-1]

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mpp;
        for (int i=0; i<nums.size(); i++){
            int num = nums[i];
            int diff = target - num;
            if (mpp.find(diff)!=mpp.end()){
                return {mpp[diff],i};
            }
            mpp[nums[i]]=i;
        }
        return {-1,-1};
    }
};